funnel/views/membership.py

```diff
@@ -0,0 +1,306 @@
+
+    route_model_map = {
+        'profile': 'project.profile.name',
+        'project': 'project.name',
+        'suuid': 'suuid',
+    }
+
+    def loader(self, profile, project, suuid):
+        membership = (
+            self.model.query.join(Project, Profile)
+            .filter(
+                Profile.name == profile,
+                Project.name == project,
+                ProjectCrewMembership.suuid == suuid,
+            )
+            .first_or_404()
+        )
+        return membership
+
+    def after_loader(self):
+        g.profile = self.obj.project.profile
+        super(ProjectCrewMembershipMixin, self).after_loader()
+
+
+@route('/<profile>/<project>/membership/<suuid>/invite')
+class ProjectCrewMembershipInviteView(
+    ProjectCrewMembershipMixin, UrlChangeCheck, UrlForView, ModelView
+):
+    __decorators__ = [legacy_redirect]
+
+    def loader(self, profile, project, suuid):
+        membership = super(ProjectCrewMembershipInviteView, self).loader(
+            profile, project, suuid
+        )
+        if not membership.is_invite or membership.user != current_auth.user:
+            raise abort(404)
+
+        return membership
+
+    @route('', methods=['GET'])
+    @render_with('membership_invite_actions.html.jinja2')
+    @lastuser.requires_login
+    def invite(self):
+        return {'membership': self.obj.current_access(), 'form': Form()}
+
+    @route('action', methods=['POST'])
+    @lastuser.requires_login
+    def invite_action(self):
+        membership_invite_form = ProjectCrewMembershipInviteForm()
+        if membership_invite_form.validate_on_submit():
+            if membership_invite_form.action.data == 'accept':
+                self.obj.accept(actor=current_auth.user)
+            elif membership_invite_form.action.data == 'decline':
+                self.obj.revoke(actor=current_auth.user)
+            db.session.commit()
+        return redirect(self.obj.project.url_for(), 303)
+
+
+@route('/<project>/membership/<suuid>/invite', subdomain='<profile>')
+class FunnelProjectCrewMembershipInviteView(ProjectCrewMembershipInviteView):
+    pass
+
+
+ProjectCrewMembershipInviteView.init_app(app)
+FunnelProjectCrewMembershipInviteView.init_app(funnelapp)
+
+
+@route('/<profile>/<project>/membership/<suuid>')
+class ProjectCrewMembershipView(
+    ProjectCrewMembershipMixin, UrlChangeCheck, UrlForView, ModelView
+):
+    __decorators__ = [legacy_redirect]
+
+    @route('edit', methods=['GET', 'POST'])
+    @render_with(json=True)
+    @lastuser.requires_login
+    @requires_roles({'editor'})
+    def edit(self):
+        previous_membership = self.obj
+        membership_form = ProjectCrewMembershipForm(obj=previous_membership)
+
+        if request.method == 'POST':
+            if membership_form.validate_on_submit():
+                previous_membership.replace(
+                    actor=current_auth.user,
+                    record_type=MEMBERSHIP_RECORD_TYPE.AMEND,
```

Comment:
The `replace` method should change the type to `AMEND`. Do not specify it in the view.